joint_trajectory_controller/test/test_trajectory_actions.cpp

```diff
@@ -0,0 +1,595 @@
+    }
+  }
+
+  void common_result_response(const GoalHandle::WrappedResult & result)
+  {
+    RCLCPP_DEBUG(
+      node->get_logger(), "common_result_response time: %f",
+      rclcpp::Clock().now().seconds());
+    common_resultcode = result.code;
+    common_action_result_code = result.result->error_code;
+    switch (result.code) {
+      case rclcpp_action::ResultCode::SUCCEEDED:
+        break;
+      case rclcpp_action::ResultCode::ABORTED:
+        RCLCPP_DEBUG(node->get_logger(), "Goal was aborted");
+        return;
+      case rclcpp_action::ResultCode::CANCELED:
+        RCLCPP_DEBUG(node->get_logger(), "Goal was canceled");
+        return;
+      default:
+        RCLCPP_DEBUG(node->get_logger(), "Unknown result code");
+        return;
+    }
+  }
+};
+
+TEST_F(TestTrajectoryActions, test_success_multi_point_sendgoal) {
+  auto traj_controller = std::make_shared<joint_trajectory_controller::JointTrajectoryController>();
+  auto ret = traj_controller->init(test_robot, controller_name);
+  if (ret != controller_interface::CONTROLLER_INTERFACE_RET_SUCCESS) {
+    FAIL();
+  }
+
+  auto traj_lifecycle_node = traj_controller->get_lifecycle_node();
+  rclcpp::Parameter joint_parameters("joints", joint_names);
+  traj_lifecycle_node->set_parameter(joint_parameters);
+
+  rclcpp::Parameter operation_mode_parameters("write_op_modes", op_mode);
+  traj_lifecycle_node->set_parameter(operation_mode_parameters);
+
+  rclcpp::executors::MultiThreadedExecutor executor;
+  executor.add_node(traj_lifecycle_node->get_node_base_interface());
+
+  traj_controller->on_configure(traj_lifecycle_node->get_current_state());
+  traj_controller->on_activate(traj_lifecycle_node->get_current_state());
+  SetUpActionClient();
+  executor.add_node(node->get_node_base_interface());
+
+  auto future_handle = std::async(
+    std::launch::async, [&executor]() -> void {
+      executor.spin();
+    });
+
+  auto thread_func = [&]() {
+      // controller hardware cycle update loop
+      auto start_time = rclcpp::Clock().now();
+      rclcpp::Duration wait = rclcpp::Duration::from_seconds(2.0);
+      auto end_time = start_time + wait;
+      while (rclcpp::Clock().now() < end_time) {
+        test_robot->read();
+        traj_controller->update();
+        test_robot->write();
+      }
+    };
+  std::thread controller_hw_thread(thread_func);
+
+  // common_goal_response and common_result_response
+  // sometimes doesnt receive calls when we dont sleep
+  std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+  GoalOptions opt;
+  opt.goal_response_callback =
+    std::bind(&TestTrajectoryActions::common_goal_response, this, _1);
+  opt.result_callback =
+    std::bind(&TestTrajectoryActions::common_result_response, this, _1);
+  opt.feedback_callback = nullptr;
+
+  // send goal
+  {
+    std::vector<JointTrajectoryPoint> points;
+    JointTrajectoryPoint point;
+    point.time_from_start.set__sec(0);  // start asap
```

Comment:
would something like this work instead?
```suggestion
    point.time_from_start = rclcpp::Time(0);  // start asap
```

Reply:
compile error:
> /home/ddeng/rmf_ws/src/ros2_controllers/joint_trajectory_controller/test/test_trajectory_actions.cpp:202:43: error: no match for ‘operator=’ (operand types are ‘trajectory_msgs::msg::JointTrajectoryPoint_<std::allocator<void> >::_time_from_start_type’ {aka ‘builtin_interfaces::msg::Duration_<std::allocator<void> >’} and ‘rclcpp::Time’)

replaced with rclcpp::Duration::from_seconds(0.0) 2b55538